src/Chapter4.hs

```diff
@@ -472,10 +487,12 @@ Implement the Applicative instance for our 'Secret' data type from before.
 -}
 instance Applicative (Secret e) where
     pure :: a -> Secret e a
-    pure = error "pure Secret: Not implemented!"
+    pure = Reward
 
     (<*>) :: Secret e (a -> b) -> Secret e a -> Secret e b
-    (<*>) = error "(<*>) Secret: Not implemented!"
+    (<*>) (Reward f) (Reward a) = Reward (f a)
+    (<*>) (Trap e) _ = Trap e
+    (<*>) _ (Trap e) = Trap e
```

Comment:
Nice! 👌🏼 
You could also reuse the `Functor` instance of `Secret` here and you would have only 2 cases 🙂